### PR TITLE
add Synology Chat plugin

### PR DESCRIPTION
**Pull request by Kolossi**
Hi Frédéric,

My first attempt at a plugin.  Synology Chat is a slack-like program available on Synology NAS devices.

I've shamelessly forked your Slack plugin, as Synology Chat uses a simplified version of the slack api - takes just text, no user or icon.  Plus they didn't make it _quite_ the same.

I'm fairly confident in what I've done but although if I drop it in my plugins directory, it shows up in plugins, the settings pages do not show up.

I'm hoping once it's listed in the plugins directory it will come to life?  If not any pointers would be much appreciated.

Regards

Paul Sweeney

### CODE REVIEW

**Comment by creecros** [comment]
Adding to the directory won't ever "Fix" your plugin, it only advertises it.
Fixed it for you, just typo's:
Kolossi/plugin-synology-chat#1

Also, your compatible versions differs from plugin to directory:
       ` "compatible_version": ">=1.2.6"`

vs

``` php
    public function getCompatibleVersion()
    {
        return '>=1.0.37';
    }
```

File: plugins.json [hunk 1]
```diff
@@ -719,6 +719,18 @@
         "remote_install": true,
         "compatible_version": ">=1.0.34"
     },
+    "synologychat": {
+        "title": "Synology Chat",
```
Comment (creecros) [inline plugins.json]: `title`, needs to match your Plugin name, which you didn't define in your plugin.php. I added it for you as `SynologyChat` so remove the space. Without the correct associations, the "up to date" function will not work properly, and users will not be able to update any future versions of your plugin from the directory.

File: plugins.json [hunk 2]
```diff
@@ -719,6 +719,18 @@
...
+        "version": "1.0.0",
+        "author": "Paul Sweeney",
+        "license": "MIT",
+        "description": "Receive individual or project notifications on Synology Chat.",
+        "homepage": "https://github.com/kanboard/plugin-slack",
+        "readme": "https://raw.githubusercontent.com/kanboard/plugin-slack/master/README.md",
+        "download": "https://github.com/Kolossi/plugin-synology-chat/releases/download/v1.0.0/SynologyChat-1.0.0.zip",
+        "remote_install": true,
+        "compatible_version": ">=1.2.6"
```
Comment (creecros) [inline plugins.json:732]: `compatible_version` differs from your plugin.php which is set at v1.0.37

**Review by creecros — COMMENTED** [review]
Couple mistakes.

File: plugins.json (hunk 2 above)
Comment (creecros) [reply plugins.json:732]: `readme` is pointing to slack plugin

Comment (creecros) [reply plugins.json:732]: `homepage` is pointing to slack plugin

**Comment by Kolossi** [comment]
> Adding to the directory won't ever "Fix" your plugin, it only advertises it.

I thought that, but couldnt' figure why the config isn't showing.  It still isn't.

> Fixed it for you, just typo's:
> [Kolossi/plugin-synology-chat#1](https://github.com/Kolossi/plugin-synology-chat/pull/1)

Thanks, merged.  Still doesn't show config item though.

> Also, your compatible versions differs from plugin to directory:
> ` "compatible_version": ">=1.2.6"`
> 
> vs
> 
> ```
>     public function getCompatibleVersion()
>     {
>         return '>=1.0.37';
>     }
> ```

yup, fixed that, and the wrong links.  They were right at some  stage but got lost along the way somehow.

**Comment by creecros** [comment]
image



This config item?

**Comment by Kolossi** [comment]
Yeah,  I see it now.  I run kanboard in docker.  I realised the plugins page was saying v0.1.0.  Seems like there was some weird caching going on.  Stopped and started the docker container and now all good.

Thanks very much for your help Craig.

**Comment by creecros** [comment]
yeah, I use docker as well, you need to restart the container for changes to take place. (if you ever figure out how to change that, you let me know!)

**Comment by Kolossi** [comment]
Ok, so at least I'm getting somewhere now.

I'll cancel this pull request for now until happy with the plugin - I'd like to add a "test" button to the config pages to test the webhook.

**Comment by Kolossi** [comment]
v1.0.0 release of the plugin created in the repo.  Now gets around Synology Chat being enthusiastic about dropping messages due to high submit rate.

**Pull request merged**